### PR TITLE
Add operator deployment infrastructure

### PR DESCRIPTION
**Pull request by theemj**

### CODE REVIEW

**Review by shaior — APPROVED** [review]

File: tests/globalhelper/deployment.go [hunk 1]
```diff
@@ -32,6 +33,18 @@ func IsDeploymentReady(operatorNamespace string, deploymentName string) (bool, e
 	return false, nil
 }
 
+// IsDeploymentInstalled checks if deployment is installed.
+func IsDeploymentInstalled(
+	cs *client.ClientSet, operatorNamespace string, operatorDeploymentName string) (bool, error) {
```
Comment (kononovn) [inline tests/globalhelper/deployment.go:38]: Why do you need cs in func parameters ?

File: tests/globalhelper/deployment.go [hunk 2]
```diff
@@ -32,6 +33,18 @@ func IsDeploymentReady(operatorNamespace string, deploymentName string) (bool, e
 	return false, nil
 }
 
+// IsDeploymentInstalled checks if deployment is installed.
+func IsDeploymentInstalled(
```
Comment (kononovn) [inline tests/globalhelper/deployment.go:37]: If you need this func only in affiliatedcertification test suite please move it under affiliatedcertification folder

File: tests/globalhelper/operator.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	"github.com/golang/glog"
+	olmv1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/test-network-function/cnfcert-tests-verification/tests/utils/namespaces"
+)
+
+func DeployOperator(namespace string, operatorGroup *olmv1.OperatorGroup, subscription *v1alpha1.Subscription) error {
```
Comment (kononovn) [inline tests/globalhelper/operator.go:15]: It's going to be used only in affiliatedcertification suite. Please move these func under affiliatedcertification folder

File: tests/utils/operator/operator.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package utils
+
+import (
+	olmv1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func DefineOperatorGroup(groupName string, namespace string, targetNamespace []string) *olmv1.OperatorGroup {
```
Comment (kononovn) [inline tests/utils/operator/operator.go:9]: Please add doc comment

File: tests/utils/operator/operator.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	return &olmv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      groupName,
+			Namespace: namespace},
+		Spec: olmv1.OperatorGroupSpec{
+			TargetNamespaces: targetNamespace},
+	}
+}
+
+func DefineSubscription(subName, namespace, channel, operatorName, catalogSource,
```
Comment (kononovn) [inline tests/utils/operator/operator.go:19]: Please add doc comment

**Review by kononovn — CHANGES_REQUESTED** [review]
Looks good. Just few comments to fix

File: tests/globalhelper/deployment.go (hunk 1 above)
Comment (theemj) [reply tests/globalhelper/deployment.go]: ah that's leftover from the inspiration code, removing

File: tests/globalhelper/deployment.go (hunk 2 above)
Comment (theemj) [reply tests/globalhelper/deployment.go]: I think we will likely need this to test the operator suite in the future https://github.com/test-network-function/test-network-function/blob/main/CATALOG.md#operator

**Review by kononovn — APPROVED** [review]
Looks good to me.

**Pull request merged**